src/main/app/servers.ts

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) 2016-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {IpcMainEvent, ipcMain} from 'electron';
+
+import {MattermostTeam, Team} from 'types/config';
+
+import {UPDATE_SHORTCUT_MENU} from 'common/communication';
+import {Logger} from 'common/log';
+import ServerManager from 'common/servers/serverManager';
+
+import ViewManager from 'main/views/viewManager';
+import ModalManager from 'main/views/modalManager';
+import MainWindow from 'main/windows/mainWindow';
+import {getLocalPreload, getLocalURLString} from 'main/utils';
+
+const log = new Logger('App.Servers');
+
+export const switchServer = (serverId: string, waitForViewToExist = false) => {
+    ServerManager.getServerLog(serverId, 'WindowManager').debug('switchServer');
+    MainWindow.show();
+    const server = ServerManager.getServer(serverId);
+    if (!server) {
+        ServerManager.getServerLog(serverId, 'WindowManager').error('Cannot find server in config');
+        return;
+    }
+    const nextTab = ServerManager.getLastActiveTabForServer(serverId);
+    if (waitForViewToExist) {
+        const timeout = setInterval(() => {
+            if (ViewManager.getView(nextTab.id)) {
+                ViewManager.showById(nextTab.id);
+                clearTimeout(timeout);
+            }
+        }, 100);
+    } else {
+        ViewManager.showById(nextTab.id);
+    }
+    ipcMain.emit(UPDATE_SHORTCUT_MENU);
+};
+
+export const handleNewServerModal = () => {
+    log.debug('handleNewServerModal');
+
+    const html = getLocalURLString('newServer.html');
+
+    const preload = getLocalPreload('desktopAPI.js');
+
+    const mainWindow = MainWindow.get();
+    if (!mainWindow) {
+        return;
+    }
+    const modalPromise = ModalManager.addModal<MattermostTeam[], Team>('newServer', html, preload, ServerManager.getAllServers().map((team) => team.toMattermostTeam()), mainWindow, !ServerManager.hasServers());
+    if (modalPromise) {
+        modalPromise.then((data) => {
+            const newTeam = ServerManager.addServer(data);
+            switchServer(newTeam.id, true);
+        }).catch((e) => {
+            // e is undefined for user cancellation
+            if (e) {
+                log.error(`there was an error in the new server modal: ${e}`);
+            }
+        });
```

Comment:
this is fine, but I wonder if it makes it easier to read by using async/await

Reply:
Yeah that's probably worth doing at some point. I think this is older code so that's why it wasn't written that way.
Will address in a follow-up PR.